tests/test_url.py

```diff
@@ -10,7 +10,7 @@
     "\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f "
 )
 _VERTICAL_COLON = "\ufe13"  # normalizes to ":"
-_FULL_WITH_NUMBER_SIGN = "\uFF03"  # normalizes to "#"
+_FULL_WITH_NUMBER_SIGN = "\uff03"  # normalizes to "#"
 _ACCOUNT_OF = "\u2100"  # normalizes to "a/c"
 
 
```

Comment:
Seeing that the same params are being used in several tests, let's just save the marker into a variable:
```suggestion

with_query_frag_preservation_matrix = pytest.mark.parametrize(
    ("original_url", "keep_query", "keep_fragment", "expected_url"),
    (
        pytest.param("http://example.com?a=b#frag", True, False, "http://example.com/test?a=b", id="query-only"),
        pytest.param("http://example.com?a=b#frag", False, True, "http://example.com/test#frag", id="fragment-only"),
        pytest.param("http://example.com?a=b#frag", True, True, "http://example.com/test?a=b#frag", id="all"),
        pytest.param("http://example.com?a=b#frag", False, False, "http://example.com/test", id="none"),
    ),
)


```

And then, use it on the tests down below as follows:
```python
@with_query_frag_preservation_matrix
def test_...(
    original_url: str,
    keep_query: bool,
    keep_fragment: bool,
    expected_url: str,
): -> None:
    ...
```

Reply:
I'm -0 for this proposal.
It reduces tests *redundancy* but on the other hand it reduces the tests *readability* as well; I should look at the variable content to figure out that parameters are used.
As a maintainer, I read the code more often than write it.
I like a moderate portion of redundancy in tests, it simplifies reading and debugging at least in *my personal* workflow. Other people could have an opposite opinion, sure.